spanner/value.go

```diff
@@ -50,6 +50,18 @@ var (
 	jsonNullBytes = []byte("null")
 )
 
+// Encoder is the interface implemented by a custom type that can be encoded to
+// a supported type by Spanner.
```

Comment:
Could we include a small example in the documentation here to show how this should be used, especially considering the necessity to implement one of the interface methods using a value and the other using a pointer.

Reply:
I have added code examples to before Encoder and Decoder. Please take a look.